scanners/noopabc.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from typing import List, Type
+
+from scanners.scannerabc import ScannerABC, ScannerABCT
+
+
+###
+# Testing scan class. Does nothing time consuming or destructive,
+# but exercises many of the main hooks of domain-scan.
+class Scanner(ScannerABC):
+
+    # CSV headers for each row of data. Referenced locally.
+    headers = ["Completed", "Constant", "Variable"]
+    # Set a default number of workers for a particular scan type.
+    # Overridden by a --workers flag.
+    workers = 2  # type: int
+
+    def __init__(self, domain: str, handles: dict, environment: dict,
+                 options: dict, extra: dict={}) -> None:
+        # Optional one-time initialization per-scan. If defined, any data
+        # will be stored in the instance's ``extra_environment`` property.
+        #
+        # Run locally.
+        logging.debug("Subclass (noopabc) __init__ method for %s." % domain)
+        self.extra_environment = {"variable": domain}
+        super().__init__(domain, handles, environment, options, extra)
+
+    def scan(self) -> dict:
+        # Required scan function. This is the meat of the scanner, where things
+        # that use the network or are otherwise expensive would go.
+        #
+        # Runs locally or in the cloud (Lambda).
+        domain = self.domain  # noqa
```

Comment:
Why assign this?

Reply:
This is a stray line from earlier testing, I'll remove it.